libs/hmpb/src/renderer.tsx

```diff
@@ -15,7 +15,7 @@ export type Page = Pick<
  * Creates a {@link RenderDocument}
  */
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-export function createDocument(page: Page) {
+export function createDocument(page: Page, cleanup: () => void) {
```

Comment:
Nit: feel free to ignore as it might be a morning review thing, but it took me a second to wrap my head around the document vs renderer cleanup with them being used close together. I could see it being helpful to rename one to make it more intuitive, maybe even keeping the name `releasePage` or `releaseLock` in this case. That being said, I understand if `cleanup` is a pattern you're trying to use.

Edit - After reading the later commits, I see why `document.cleanup()` makes sense. So, if anything I'd consider renaming `renderer.cleanup()` to `.close()` or `.shutdown()`, but fine as is

Reply:
Good feedback - I'll rename one of them to clarify the difference